driver-core/src/main/com/mongodb/client/model/mql/MqlExpression.java

```diff
@@ -953,6 +953,15 @@ public MqlString asString(final MqlString timezone, final MqlString format) {
                 .append("timezone", toBsonValue(cr, timezone))));
     }
 
+    @Override
+    public MqlString asString(final MqlString timezone) {
+        Assertions.notNull("timezone", timezone);
+        return newMqlExpression((cr) -> astDoc("$dateToString", new BsonDocument()
+                .append("date", this.toBsonValue(cr))
+                .append("timezone", toBsonValue(cr, timezone))));
+    }
+
+
```

Comment:
If my other comments are correct, I think we might instead make the following changes in this PR:

```suggestion
    public MqlString asString(final MqlString timezone) {
        fail(); // intentionally not implemented, see DRIVERS-2620
    }

    public MqlString asString() {
        fail(); // intentionally not implemented, see DRIVERS-2620
    }

```

Reply:
I think we need to decide whether to close this PR in the first place. If we want to merge this PR, I would commit your suggestion, together with necessary testing changes.

Reply:
What would the testing changes be?

Reply:
if we changed as suggested, the newly-added `dateAsStringWithDefaultFormat()` would fail invariably. Then we might need to remove it totally and replace it with some simple testing cases to assert exception is thrown?

Reply:
nvm. as we discussed, if we adopted the above change, no testing is required.

Reply:
> If my other comments are correct, I think we might instead make the following changes in this PR:

I committed the suggestion and found the following issue:

- `MqlExpression` has `asString()` method already, which ends up with `$toString` expression, not the `$dateToString` expression. It was covered in the existing testing case `com.mongodb.client.model.mql.TypeMqlValuesFunctionalTest#dateAsStringTest()`

Not sure whether further testing is needed; for reproduce the bug on v6, we need to do the following steps:

1. restore the original implementation (or not `fail()`)
2. use assumption of v6 instead of v7.1+

So given our current `throw fail();` implementation, there is no way to test the bug on v6 using `assertExpression()`.